src/fetch_candidates.php

```diff
@@ -0,0 +1,70 @@
+<?php
+session_start();
+include_once str_replace('/', DIRECTORY_SEPARATOR, 'includes/classes/file-utils.php');
+require_once FileUtils::normalizeFilePath('includes/classes/db-connector.php');
+require_once FileUtils::normalizeFilePath('includes/session-handler.php');
+require_once FileUtils::normalizeFilePath('includes/classes/session-manager.php');
+include_once FileUtils::normalizeFilePath('includes/error-reporting.php');
+
+// Initialize candidates array
+$candidates = [];
+
+if (isset($_SESSION['voter_id']) && ($_SESSION['role'] == 'admin' || $_SESSION['role'] == 'head_admin')) {
+
+    // ------ SESSION EXCHANGE
+    include FileUtils::normalizeFilePath('includes/session-exchange.php');
+    // ------ END OF SESSION EXCHANGE
+
+    // Database connection
+    $connection = DatabaseConnection::connect();
+
+    if ($connection) {
+        $voter_id = $_SESSION['voter_id'];
+
+        if (isset($_SESSION['organization']) && isset($_GET['position_id']) && isset($_GET['election_year'])) {
+            // Retrieve the organization name
+            $organization = $_SESSION['organization'];
+
+            // Fetch election years
+            $yearsQuery = "SELECT DISTINCT election_year FROM candidate ORDER BY election_year DESC";
+            $result_years = $connection->query($yearsQuery);
+        
+            $position_id = $_GET['position_id'];
+            $election_year = $_GET['election_year'];
+
+            // Prepare and execute query
+            $query = $connection->prepare("SELECT c.first_name, c.last_name, COUNT(v.vote_id) as vote_count
+                                        FROM candidate c
+                                        JOIN vote v ON c.candidate_id = v.candidate_id
+                                        WHERE c.position_id = ? AND c.election_year = ?
+                                        GROUP BY c.candidate_id
+                                        ORDER BY vote_count DESC");
```

Comment:
Do you have a new column `election_year` in candidate table?

Reply:
[SQLResult.txt](https://github.com/user-attachments/files/15522171/SQLResult.txt)
SQL used in the result page